appinventor/lib/blockly/src/core/block.js

```diff
@@ -140,6 +140,7 @@ Blockly.Block.prototype.fill = function(workspace, prototypeName) {
   this.inputsInline = false;
```

Comment:
The core changes are as follows : 
Add an undefined property to blocks.
Allow blocks to be set as defined and undefined (visual + behavioral change) 
Allow blocks to be marked/unmarked bad. (only visual change)